### DISCUSSION TITLE
Optimization Parameters

### DISCUSSION THREAD

**Discussion by dlcfort** [giaf/hpipm#150] [Q&A] [answered]
Do you have any documentation that helps explain each of the argument parameters?

Such as:
iter_max, alpha_min, mu0, tol_stat, tol_eq, tol_ineq, tol_comp, reg_prim, reg_dual, warm_start, pred_corr, cond_pred_corr, comp_res_pred, comp_res_exit, lam_min, t_min, split_step, t_lam_min.

Thank you

**Comment by giaf** [comment, marked as answer]
As of now there is the description in the header file 

OCP QP: https://github.com/giaf/hpipm/blob/master/include/hpipm_d_ocp_qp_ipm.h#L58

dense QP: https://github.com/giaf/hpipm/blob/master/include/hpipm_d_dense_qp_ipm.h#L60

**Reply by dlcfort** [reply]
Is there a definition for the following parameters? I could not find comments for the following parameters.

tol_stat
tol_eq
tol_ineq
tol_comp

Thank you

**Comment by giaf** [comment]
These are the exit tolerances for the IPM iterations.
At each IPM iteration, the residuals of the linearized KKT conditions are computed, and if the infinity norm of these residuals are all smaller than these tolerances, the solution is considered found and returned.
The 4 linearized KKT conditions are equations on stationarity, equality constraints feasibility, inequality constraints feasibility and complementarity.
